Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -51,8 +97,47 @@ public static int[] RemoveFirst(int[] array)
         /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
-            return array;
+            var newArray = ProcessArray(array, index);
+            return newArray; 
+        }
+
+        private static int[] ProcessArray(int[] array, int index)
+        {
+            var newArray = CheckForInValidCases(array, index);
+            newArray = Remove(array, index);
+            return newArray;
+        }
+
+        private static int[] CheckForInValidCases(int[] array, int index)
+        {
+            if (array == null || index < 0 || index == array.Length || array.Length == 0 || array == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'false'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4887669)